test/e2e/util.go

```diff
@@ -653,3 +653,75 @@ func assertDenyLogs(targetNodeName string, namespace string, policyName string,
 	}
 	return false, nil
 }
+
+// patchService patches service serviceName in namespace serviceNamespace.
+func patchService(c kubernetes.Interface, serviceName, serviceNamespace, jsonPath, value string) error {
+	patch := []struct {
+		Op    string `json:"op"`
+		Path  string `json:"path"`
+		Value string `json:"value"`
+	}{{
+		Op:    "replace",
+		Path:  jsonPath,
+		Value: value,
+	}}
+	patchBytes, _ := json.Marshal(patch)
+
+	_, err := c.CoreV1().Services(serviceNamespace).Patch(
+		context.TODO(),
+		serviceName,
+		types.JSONPatchType,
+		patchBytes,
+		metav1.PatchOptions{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Poll a Service until the number of its ClusterIps is larger equals the desired number.
+func waitForServiceClusterIpsNum(c kubernetes.Interface, serviceName, serviceNamespace string, number int) error {
```

Comment:
I hoped that there is a way to check when service changes are actually applied, but I think there is no such way.
So this check doesn't do anything since updated service with 2 clusterIPs is a response to update request => no need to wait for it.
We rely on low latency in these tests, so you can just remove this check.